addons/test/misra/misra-test.c

```diff
@@ -337,21 +337,21 @@ const misra_8_1_a; // 8.1 8.4
 static int misra_8_2_a (int n, ...);
 extern int misra_8_2_b (int n);
 extern int misra_8_2_c (int); // 8.2
-static int misra_8_2_d (); // 8.2
+static int misra_8_2_d (); // TODO: 8.2
```

Comment:
I guess we don't warn for missing void anymore that would be unfortunate.. sorry I missed this detail before.

Reply:
Yea it is unfortunate, for a proper fix there are a couple of solutions:

* Use rawTokens to check for this although you cant use the symboldatabase to find the start of a function declaration, it should be easy to find in C since all functions are declared at global scope
* Add some flag(similar to how we remove other keywords like `constexpr`, `typename`, etc) and check for that flag
* Link tokens back to rawTokens so a checker could just say `Token::Match(tok->rawToken(), "( void )")`

Reply:
Such stylistic checks would be similar to a check for braces `{}` when using if/while/for(some coding guidelines require this). In cppcheck, we canonicalize the tokens to always have braces so we cant check this using the normalized tokenlist. In an addon, a user should use the rawTokens for this check, it doesn't make sense for us to remove such simplifications, as it can break many assumptions in code and it makes future checks more complicated as it needs to check two scenarios.

I think the same principle applies to void parameters as well.

Reply:
I don't understand why the `void` is removed? 

For C++ it's just redundant so I am not strongly against that.

But for C it's important and could influence how function pointers should be set for instance.

Reply:
> There are too many places in the code where we assume a void parameter function is a unary function(not just in the templatesimplifier), which affects both C and C++ analysis.

ok I didn't read this.. I really think there should be a proper solution for misra. but I can look at that later.